src/CLI/Command/Check.php

```diff
@@ -120,7 +122,8 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             $phpVersion = $input->getOption('target-php-version');
             $targetPhpVersion = TargetPhpVersion::create($phpVersion);
 
-            $progress = $verbose ? new DebugProgress($output) : new ProgressBarProgress($output);
+            $defaultProgressBar = (new CiDetector())->isCiDetected() ? new VoidProgress() : new ProgressBarProgress($output);
+            $progress = $verbose ? new DebugProgress($output) : $defaultProgressBar;
```

Comment:
Is it better to move these two lines into a specific factory / class and increase the readability of this code? 
What do you think @fain182 ?

Reply:
I tried, but it was worse...

So I simplified this part moving some logic inside the ProgressBarProgress, what do yout think? it's better now or before?